src/protocol/http/org/apache/jmeter/protocol/http/proxy/DefaultSamplerCreator.java

```diff
@@ -389,6 +390,9 @@ protected String computeContentEncoding(HttpRequestHdr request,
             if (pageEncodings != null) {
                 synchronized (pageEncodings) {
                     contentEncoding = pageEncodings.get(urlWithoutQuery);
+                    if(contentEncoding == null && pageEncodings.size()>0) {
```

Comment:
`Map` has a new method `getOrDefault`, which could be used to set a default value, but are you sure it is a good idea to set the first value of pageEncodings values? Those values come from a map and are not sorted.
This would seem like a heuristic that works when all pages have the same encoding, but is fragile in all other circumstances.
Maybe add a parameter - I honestly don't know where - to give the user the chance to specify a default encoding.

Reply:
I agree with you. There is an option to provide "content encoding" for HTTP request in JMeter tool itself. could you please provide some information related to class, method or variable name from where we can get the "content encoding" provided by user in JMeter tool for HTTP request, so that we can use the same in our method.

Reply:
Sorry for delay in providing the response, since we were considering the approach provided by you.

@FSchumacher, Thanks for checking the PR request.
>Map has a new method getOrDefault, which could be used to set a default 
>value, but are you sure it is a good idea to set the first value of pageEncodings values?
>This would seem like a heuristic that works when all pages have the same encoding, but is fragile in all other circumstances.
>Maybe add a parameter - I honestly don't know where - to give the user the chance to specify a default encoding.

Thank you for your valuable suggestion. I agree with you for using "getOrDefault" method in such scenario.
If default encoding is provided by user, then using "getOrDefault" method can solve the problem.

We agree with you that it cannot solve if multiple applications of different encoding is included.
We have assumption that pages with same domain have same encoding is very common and pages with same domain having different encoding is very rare scenario.
Please refer to below example scenarios for details:
1. Pages with same base domain (abc.com), have same encoding.
	abc.com/index.html : shift_jis 
	abc.com/first.html : shift_jis

2. Pages with same base domain (xyz.com), have different encoding.
	xyz.com/index.html : shift_jis
	xyz.com/first.html : euc-jp

If our assumption is correct, we provide the following solution:

1. If pageEncodings contains key with page url (abc.com/first.html), then value of encoding will be used.
2. If pageEncodings does not have the key with page url, then search for base domain(currently it does not store, it can be implemented) key and use the encoding. 
3. If pageEncodings does not have the key with base domain, 
3.1 If user has provided default value of encoding, it should be used.
3.2 If user has not provided default encoding, UTF-8 can be used.

Is it possible, based our assumption, can this solution be used for the current problem.
So that we can provide the source code for the solution.

If this approach can not be adopted, then we can go for another approach.
We can request user to provide config file containing domain level or page level encoding.


Please share your valuable feedback, which solution is better and can be accepted in JMeter, so that we can proceed further for this bug.